<?xml version="1.0" ?>
<!-- =================================================================================== -->
<!-- |    This document was autogenerated by xacro from xacro/sr_arm_motor.urdf.xacro  | -->
<!-- |    EDITING THIS FILE BY HAND IS NOT RECOMMENDED                                 | -->
<!-- =================================================================================== -->
<robot name="sr_arm_motor" xmlns:xacro="http://ros.org/wiki/xacro">
  <link name="shadowarm_base">
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <mass value="0.4"/>
      <inertia ixx="1.0" ixy="0.0" ixz="0.0" iyy="0.0" iyz="0.0" izz="0.0"/>
    </inertial>
    <visual>
      <origin rpy="1.57079632679 0 0" xyz="0 0 0.005"/>
      <geometry name="shadowarm_base_visual">
        <mesh filename="package://sr_hand/model/meshes/arm/arm_base.mesh" scale="0.001 0.001 0.001"/>
      </geometry>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0.005"/>
      <geometry name="shadowarm_base_collision">
        <box size="0.03 0.03 0.01"/>
      </geometry>
    </collision>
  </link>
  <link name="shadowarm_trunk">
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0.27025"/>
      <mass value="5"/>
      <inertia ixx="1.0" ixy="0.0" ixz="0.0" iyy="0.0" iyz="0.0" izz="0.0"/>
    </inertial>
    <visual>
      <origin rpy="1.57079632679 3.14159265359 1.57079632679" xyz="0 0 0.0165"/>
      <geometry name="shadowarm_trunk_visual">
        <mesh filename="package://sr_hand/model/meshes/arm/arm_trunk.mesh" scale="0.001 0.001 0.001"/>
      </geometry>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0.27025"/>
      <geometry name="shadowarm_trunk_collision">
        <box size="0.03 0.03 0.5405"/>
      </geometry>
    </collision>
  </link>
  <joint name="trunk_rotation" type="revolute">
    <parent link="shadowarm_base"/>
    <child link="shadowarm_trunk"/>
    <origin rpy="0 0 0" xyz="0 0 0.01"/>
    <axis xyz="0 0 1"/>
    <limit effort="10" lower="-1.57079632679" upper="0.785398163397" velocity="1.0"/>
    <dynamics damping="200.5"/>
  </joint>
  <link name="shadowarm_upperarm">
    <inertial>
      <origin rpy="0 0 0" xyz="0.1845 -0.067 0"/>
      <mass value="2"/>
      <inertia ixx="1.0" ixy="0.0" ixz="0.0" iyy="0.0" iyz="0.0" izz="0.0"/>
    </inertial>
    <visual>
      <origin rpy="-1.57079632679 0 -1.57079632679" xyz="0 0.067 0"/>
      <geometry name="shadowarm_upperarm_visual">
        <mesh filename="package://sr_hand/model/meshes/arm/upper_arm.mesh" scale="0.001 0.001 0.001"/>
      </geometry>
    </visual>
    <collision>
      <origin rpy="0 1.57079632679 0" xyz="0.1845 0 0"/>
      <geometry name="shadowarm_upperarm_collision">
        <box size="0.03 0.03 0.369"/>
      </geometry>
    </collision>
  </link>
  <joint name="shoulder_rotation" type="revolute">
    <parent link="shadowarm_trunk"/>
    <child link="shadowarm_upperarm"/>
    <origin rpy="0 0 0" xyz="0 -0.134 0.5405"/>
    <axis xyz="0 1 0"/>
    <limit effort="10" lower="0" upper="1.57079632679" velocity="1.0"/>
    <dynamics damping="50.5"/>
  </joint>
  <link name="shadowarm_lowerarm">
    <inertial>
      <origin rpy="0 0 0" xyz="0.0655 0 -0.08"/>
      <mass value="1.7"/>
      <inertia ixx="1.0" ixy="0.0" ixz="0.0" iyy="0.0" iyz="0.0" izz="0.0"/>
    </inertial>
    <visual>
      <origin rpy="3.14159265359 0.5 0" xyz="0.036 0 -0.03"/>
      <geometry name="shadowarm_lowerarm_visual">
        <mesh filename="package://sr_hand/model/meshes/arm/elbow_flex.mesh" scale="0.001 0.001 0.001"/>
      </geometry>
    </visual>
    <collision>
      <origin rpy="0 1.57079632679 0" xyz="0.0655 0 -0.08"/>
      <geometry name="shadowarm_lowerarm_collision">
        <box size="0.03 0.03 0.131"/>
      </geometry>
    </collision>
  </link>
  <joint name="elbow_abduction" type="revolute">
    <parent link="shadowarm_upperarm"/>
    <child link="shadowarm_lowerarm"/>
    <origin rpy="0 0 0" xyz="0.369 0 0"/>
    <axis xyz="0 -1 0"/>
    <limit effort="10" lower="0" upper="2.09439510239" velocity="1.0"/>
    <dynamics damping="50.5"/>
  </joint>
  <link name="shadowarm_handsupport_motor">
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <mass value="0.75"/>
      <inertia ixx="1.0" ixy="0.0" ixz="0.0" iyy="0.0" iyz="0.0" izz="0.0"/>
    </inertial>
    <visual>
      <origin rpy="0 0 -1.57079632679" xyz="-0.04 0 0"/>
      <geometry name="shadowarm_handsupport_visual">
        <mesh filename="package://sr_hand/model/meshes/arm/elbow_rotate_motor.mesh" scale="0.001 0.001 0.001"/>
      </geometry>
    </visual>
    <collision>
      <origin rpy="0 1.57079632679 0" xyz="0 0 0"/>
      <geometry name="shadowarm_handsupport_collision">
        <cylinder length="0.001" radius="0.06"/>
      </geometry>
    </collision>
  </link>
  <joint name="forearm_rotation" type="revolute">
    <parent link="shadowarm_lowerarm"/>
    <child link="shadowarm_handsupport_motor"/>
    <origin rpy="0 0 0" xyz="0.131 0 -0.08"/>
    <axis xyz="1 0 0"/>
    <limit effort="10" lower="-1.57079632679" upper="1.57079632679" velocity="1.0"/>
    <dynamics damping="10.5"/>
  </joint>
</robot>
